### DISCUSSION TITLE
one big problem in ajax requests

### DISCUSSION THREAD

**Discussion by m-jooibar** [apache/superset#17368] [Q&A / Help]
Hello Superset team .
I run superset by docker-compose-non-dev.yml file.
I saw a strange case in my dashboard. I created a dashboard and I imported one chart in it . When I load my dashboard I have several xhr requests in my google chrome network section.
as you can see in bellow photo we have several xhr request :

![](https://user-images.githubusercontent.com/42513136/140608726-eb81356b-a838-4a75-9b1b-21fc40fbe73d.jpg)

i expand two of them for you :

First : datasets
as you can see you give some details that are not necessary for all users. why user can see important details such as table names,columns names and our queries ? I searched in your docs but I did not find any .
I guessed maybe if I run superset by docker-compose-non-dev it will run on production mode and we will not see this information but my guess was wrong .

![](https://user-images.githubusercontent.com/42513136/140608736-94fb6d0a-8fc2-4746-98fd-3853e801c13f.jpg)

Second: http://localhost:8088/api/v1/chart/data?form_data=%7B%22slice_id%22%3A1%7D&dashboard_id=1
In this case we have a previous problem too . We are seeing some data that should be hidden. i know that we can handle this by result_type parameter that it can give on of these values [ columns,full,query and ... ]
but i do not know how we can pass this value to this parameter when we are loading dashboard in Superset or how we can set 'columns' by default for result_type

![](https://user-images.githubusercontent.com/42513136/140608742-69422d8f-16fb-4c9d-9e1c-73d2a02e5d10.jpg)

I hope you answer me . I am waiting for your answer. Thanks .

**Comment by srinify** [comment]
FYI this is the current thread for discussing this for anyone who's interested:

https://github.com/apache/superset/issues/17371
